src/main/java/store/controller/StoreController.java

```diff
@@ -0,0 +1,205 @@
+package store.controller;
+
+import java.util.List;
+import store.constant.Discount;
+import store.constant.Message;
+import store.model.ProductBox;
+import store.model.ShoppingItem;
+import store.model.StockInventory;
+import store.model.PromotionCatalog;
+import store.model.ShoppingCart;
+import store.service.StoreService;
+import store.validator.Validator;
+import store.view.InputView;
+import store.view.OutputView;
+
+public class StoreController {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final StoreService storeService;
+
+    public StoreController(InputView inputView, OutputView outputView, StoreService storeService) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.storeService = storeService;
+    }
+
+    public void run() {
+        PromotionCatalog promotionCatalog = storeService.createPromotion();
+        StockInventory stockInventory = storeService.createInventory(promotionCatalog);
+        ShoppingCart shoppingCart = new ShoppingCart();
+        while (true) {
+            outputView.displayInventory(stockInventory);
+            purchaseProduct(stockInventory, shoppingCart);
+            receiptHandler(stockInventory, shoppingCart);
+            if (!askToContinueShopping()) {
+                return;
+            }
+        }
```

Comment:
while에 true를 넣는 것은 지양해야한다고 알고 있습니다. 조금만 실수하면 무한 반복이라는 치명적인 문제가 생기기 때문입니다. 저는 코드리뷰 해드릴때 이 부분에서 do - while문을 추천드립니다
```suggestion
        do {
            outputView.displayInventory(stockInventory);
            purchaseProduct(stockInventory, shoppingCart);
            receiptHandler(stockInventory, shoppingCart);
        } while (askToContinueShopping());
```